### PR TITLE
rapl: Added RAPL component support for Intel RaptorLake.

### PR DESCRIPTION
**Pull request by willowec**
## Pull Request Description

This PR adds support for the `rapl` component on Intel's RaptorLake architecture. Adding this was extremely simple as RaptorLake has the same RAPL setup as SandyBridge and IvyBridge.

Tested systems:

| **System Name** | **Family/Model/Stepping** | `papi_native_avail` | `papi_component_avail` | `papi_command_line` | **RAPL tests** |
|-------------|-----------------------|-------------------|----------------------|-------------------|------------|
| Raptor Lake-S/HX B0  | 0x6/0xb7/0x1 |<ul><li> [x] </li></ul>|<ul><li> [x] </li></ul> | <ul><li> [x] </li></ul> |<ul><li> [x] </li></ul>|

## Author Checklist
- [X] **Description**
_Why_ this PR exists. Reference all relevant information, including _background_, _issues_, _test failures_, etc
- [X] **Commits**
_Commits_ are self contained and only do one thing
_Commits_ have a header of the form: `module: short description`
_Commits_ have a body (whenever relevant) containing a detailed description of the addressed problem and its solution
- [X] **Tests**
The PR needs to pass all the tests

### CODE REVIEW

**Comment by Treece-Burgess** [comment]
@willowec How many different CPU's in the Intel RaptorLake family were you able to test on for this change and what exactly did you test?

It would be good if you could test multiple different CPU's in the Intel RaptorLake family and after building the RAPL component, running the utilities: `papi_native_avail`, `papi_component_avail`, and `papi_command_line` to make sure you get the expected results. As well as the tests in `rapl/tests/`.  See PR #275 for how I went about testing to add support for AMD family 25.

**Comment by willowec** [comment]
@Treece-Burgess I've added the table of tests that I have run, but I only have access to one of the three varieties of RaptorLake machines. I'm glad you pointed this out though, because I when I created this PR I did not know that multiple varieties existed. Would you like me to add the other two varieties without testing them?

**Comment by Treece-Burgess** [comment]
@willowec If you could add the other two varieties that would be great! Two other comments:

1. Could you order the cases' in ascending order? E.g.:
```
case 42:
case 58:
case 183:
...
```
2. In your commit message can you add which RaptorLake's you were able to test on (including the Family/Model/Stepping) and which you were not able to test on (does not need to include the Family/Model/Stepping)? Such that we can use this as a reference later if needed.

File: src/components/rapl/linux-rapl.c [hunk 1]
```diff
@@ -405,6 +405,9 @@ _rapl_init_component( int cidx )
 
 		case 42:	/* SandyBridge */
 		case 58:	/* IvyBridge */
+		case 183:	/* RaptorLake-S/HX */
+      case 186:	/* RaptorLake */
```
Comment (Treece-Burgess) [inline src/components/rapl/linux-rapl.c:409]: Hello @willowec can you align these case's and rebase? 

I will approve and merge after that.

**Review by Treece-Burgess — CHANGES_REQUESTED** [review]

File: src/components/rapl/linux-rapl.c (hunk 1 above)
Comment (willowec) [reply src/components/rapl/linux-rapl.c]: Done. I'll make sure to change my editor to use tabs for indentation instead of spaces. Thank you!

Comment (Treece-Burgess) [reply src/components/rapl/linux-rapl.c]: I think some files use tabs while others use spaces.  At least that is what I have noticed when making changes throughout the library. Thank you for the changes!

**Review by Treece-Burgess — APPROVED** [review]

**Pull request merged**